### ISSUE TITLE
What to do with manual installation of bionics?

### ISSUE THREAD

**Issue by Night-Pryanik** [CleverRaven/Cataclysm-DDA#24113]
The series of PRs concerning autodocs and bionics installation overhaul have been merged some time ago. The bionics manipulation overhaul is almost done, and now we can sum up the results. 

- Bionics can be (un)installed through the autodoc. Bionics (for now) still can be installed old-style manual way.
- Autodocs spawn (for now) in three places: hospital, bionic basement, and the military bunker's basement;
- Patient need to be induced into anesthesia so autodoc can conduct an operation upon him;
- Anesthesia spawn (for now) in five places: hospital, bionic basement, doctor's office, veterinarian office, and the military bunker's basement;
- To install bionics, patient no longer needs to achieve 10 percent painkiller threshold;
- To uninstall bionics, patient no longer needs the scalpel and a first aid kit.

I think this all has prepared the ground for the final step. The only remaining question is: what to do with manual installation of bionics? We could:

- Remove the manual installation altogether so autodoc will be the only way to install bionics. This include complete removing of `install_bionic_actor` from the code;
- Remove only `install_bionic` use action, and leave actor. This way someone someday could add a mod that will return the manual installation into the game as an alternative;
- Leave manual installation in the game and rework the autodocs so that they should have a higher chance of successful operation. This require thought of how manual installation should interact with anesthesia.

**Comment by ZhilkinSerg** [comment]
I suggest making a game option to allow manual installation/removal of bionics and move it to mod.

**Comment by ZhilkinSerg** [comment]
We also need some kind of advertisement, so new players would know about autodocs - flyers, journals, computer entries, npc advices, or something.

**Comment by Rivet-the-Zombie** [comment]
I kinda like the idea of being able to manually install them with all the gory possibilities that failure can produce. That said, it should be much, much safer to just use the autodoc thing instead.

**Comment by papersplease** [comment]
Have different tiers autodocs install different tier bionics, for example. Manual installation would have been fine for the simplest CBMs while under new anesthetics, but as I understand it, they render you unconscious. So audotocs are pretty much the only logical option IMO. You can keep the skill requirement for the installation as well, since autodocs are intended to be operated by some qualified medical staff.

I don't think there's a pressing need to remove anything from the code entirely, is it? At least for the testing period, you can keep it to be able to revert something if any issues arise.

**Comment by kevingranade** [comment]
I'm in favor of leaving the code in place to allow self-installation, but in the main game your options are autodoc or a npc doctor (or another TBD third party).
This can either be as an option, or as an "installation kit" type item, or potentially something else.  I don't have a strong opinion about how it should work, I'm just pointing out that it can work the way the mod maintainer wants it to, and there are options for how to go about it.

The other question is, is it reasonable right now to expect players to discover the existence of autodocs and npc doctors?  I could be missing something, but it doesn't seem like it.  Advertisements and NPC dialog are good options, the existence of broken autodocs would be a good pointer, and an autodoc-related quest would be great.  
Likewise we might consider adding a copy of the doctor questline from the ranch to the survivor center, and one to the new base construction feature.

I think several of these options (not mecessarally all of them though) need to happen before we turn self-installation off by default.

**Comment by nexusmrsep** [comment]
In my opinion few of CBMs that don't require extreme surgical incursion for proper instalation might be flagged as possible for self-instalation. Example: power banks. We can discuss which ones those would be. Rest, aspecialy including limb and/or organ replacement, should be undoubtly locked for self instalation. Self instalation would still remain risky even for rudimental cbm's like power banks. So I am for leaving those part of code avalaible.

**Comment by AMurkin** [comment]
Idea: install power banks of each kind without autodoc, except the first one.

**Comment by ZhilkinSerg** [comment]
We have bionic slots project, so autodoc can be used to create these kind of slots. Then some simple bionics could be installed to these slots (e.g. powerbanks), but if you want to expand slots (to make room for more bionics) or install some complex bionics (e.g. artificial eye) - you will need autodoc.

**Comment by TechyBen** [comment]
Those seem some good ideas. As the CMBs could be entirely surgical, or some could be nano/injection/slot/switch out style. Any system could require full per patience adjustments (anti rejection drugs, adjusted to blood type/DNA/shape of patient), or could be universal.

So as a player, I'm happy for either, as long as the description and function are clear in the game (mechanics and implementation and/or descriptions). So I know what each thing is or is not.

For example, armour plating would have to be shaped to each person. Does the CMB do this, or the autodoc? Is a mechanical replacement limb grafted, or just applied to a grafted socket (thus first install needing autodoc, additional would be easy switch outs). Do the nano bots need programming? If not, why can I not just ingest/inject them and they work? As a player, I feel frustrated if the game is not clear on what is happening. I need more than "you can't do that", and closer to "*why* you can't do that" messages in game.

So really, the possibilities are limitless. It's left to an artistic or gameplay decision. Are the CMBs simple (artistic and realistic decision) thus similar to current prosthetics needing full surgical applications and anti rejection drugs. Or are the future tech with nano/DNA encompassing self installing packs?

To add to this, IMO no "salvaged" CMBs would work at all. There is little to no way to successfully get one out functioning, let alone re-introduce it to another person! However, that is for another discussion.

**Comment by BrianLefler** [comment]
- Move Bionic self-installation to the crafting menu under a Surgery tab.  Make it take time, require thread, disinfectant, bandages, sewing kit, scalpel, scissors.  Instead of crafting an item at the end, the CBM gets installed.  The tab makes this discoverable.  There is still a risk chance. (poor man's version: no crafting, but make the action require a surgical kit).
- Autodocs as a safer alternative with a higher success rate (0% for the simplest stuff).
- Trivial and Invasive Bionics get moved into different categories --invasive ones can't be self-installed.  Autodocs can autoinstall trivial stuff, but requires the first aid skills for invasive bionics, making invasive bionics late-game without cutting the player out of early bionics.  Perhaps indicate this with a clear naming scheme, like trivial CBMs become ABMs (Automatic Bionic Modules) or maybe large ones become DBMs or LBMs (Disruptive BMs, Large BMs).  Maybe invasive bionics finally use the slot system, and trivial ones are unlimited.
- Have power storage not require surgery (instead, require a charging CBM to exist before installation).
- Salvaged CBMs are (variably) damaged, increasing install risk. Repairable with high electronics.  Autodoc auto-install only works on perfect quality CBMs.

I would be happy to do the development on self-installation via crafting menu if the idea is endorsed.

**Comment by Night-Pryanik** [comment]
> Make it take time, require thread, disinfectant, bandages, sewing kit, scalpel, scissors.

No anesthesia or painkillers?

**Comment by BrianLefler** [comment]
Oh yeah painkillers.  Which might be nice in explicitly defining requirements (say 2 codeine or 1 heroin or 1 morphine), though it wouldn’t interact meaningfully with taking actual painkillers.

I haven’t used an auto doc yet but am really nervous about anesthesia as described.  Does anesthesia mean a death sentence if you clear the area but get unlucky with a zombie hulk wander spawn?  Because they could break in and kill you without waking you?

**Comment by 1skandar** [comment]
Noting that a CBM needs an autodoc or Doctor NPC to install in the description of the CBM itself would be the most unmissable way of making sure the players are aware.

**Comment by Night-Pryanik** [comment]
> Noting that a CBM needs an autodoc or Doctor NPC to install in the description of the CBM itself would be the most unmissable way of making sure the players are aware.

Yes, I was thinking about it in the first place. Also add info in the help menu.

**Comment by BrianLefler** [comment]
>> Does anesthesia mean a death sentence if you clear the area but get unlucky with a zombie hulk wander spawn? Because they could break in and kill you without waking you?

> Yes, it does.

Is there a solution to that?  I realize this is unlikely, but I would not enjoy unavoidably dying from bad luck, when I was using the equipment as intended.  It's part of the game to die from going to sleep in a safe building and waking up with zombies in your lap, but at least people get a chance to run/fight.

And agree, a crafting menu doesn't make sense.  Maybe you could get similar effect by having a kit crafted once per surgery (the tools would be returned after surgery).  Then installing a bunch of bionics wouldn't be too bad--someone can  just type '-' to craft the previous item between installs, until they ran out of drugs or first aid kits.

- Prepped surgical kit: "A complete surgical kit prepared with everything you need for elective surgery and a full recovery: sterilized tools, medical supplies, and the industrial strength painkillers.  Use this item to install a CBM.  This process takes an hour."  Proposed components: 1 scalpel, 1 scissors, 1 syringe, 2x first aid kit, 1 morphine/heroin/oxycotton, 50 thread/sinew (tool with sewing 1).
- Improvised surgical kit: "A makeshift surgical kit with enough cobbled together supplies to perform dangerously elective surgery.  Use this item to install a CBM with a decreased chance of success.  This process takes an hour and inflicts damage."  Proposed components: 1 scalpel/xacto, 10 medical tape or 20 duct tape, 10 aspirin or 2 codeine, +ingredients similar to an improvised first aid kit.

**Comment by Night-Pryanik** [comment]
> Is there a solution to that?

Exiting from anesthesia IRL is a long and complicated process, not like almost instant waking up from a sleep. For now we're leaving this effect behind the curtains, but can add the gradual exiting from anesthesia (like adding some "Dizziness" effect) in the future. Even if autodoc wakes you up right on the moment monster started attacking you, I thought that you just won't have enough time to fully wake up and take your chances of fighting or fleeing.

**Comment by Night-Pryanik** [comment]
> That said, it should be much, much safer to just use the autodoc thing instead.

We could leave the manual installation in the game (though it will certainly require major rework in terms of painkillers/anesthesia and time consume) with all the checks for player skills, while autodoc could use fixed 20 level in all required skills (as NPC doctor is doing right now). This way could provide the natural choice towards the autodoc.

**Comment by Night-Pryanik** [comment]
> all you need is a clean cut throu the skin to expose the joint and both bones to clamp the ratchet, and then sew the skin together

You forgot about the wiring required to connect Joint Torsion Ratchet to the power banks somewhere inside one's body (where are they located, by the way?). Having that in mind, we could exclude all bionics with `BIONIC_POWER_SOURCE` flag from the list of easy-to-install bionics.

So, bionics that occupy both hands or both arms, bionics that involve inter-head or eye work, bionics that require complex operation on one's intestines, bionics that require limb replacement, bionics that are bionic power source - I think all of these are off-limits to self-installation. What remains? Terranian Sonar and a number of bionics that can be embedded in one hand or fingers? I wonder if this small list justifies existence of manual self-installation?

**Comment by nexusmrsep** [comment]
< You forgot about the wiring required to connect Joint Torsion Ratchet to the power banks somewhere inside one's body (where are they located, by the way?). Having that in mind, we could exclude all bionics with BIONIC_POWER_SOURCE flag from the list of easy-to-install bionics.

In that case, its actualy reasonable to ditch self-instalation in vanilla all in all & leave the code for modders.

Unless a power network cbm would be taken into consideration, but I don't like that idea myself, and mention it only for the sake of the discussion. It would unnecesarily introduce layers on layers of CBM and interdependency, and with access to autodoc to install the network, there is no incentive to revert to self instalation for powered CBMs.

**Comment by FulcrumA** [comment]
I am absolutely fine with leaving manual installation option as a possibility, just one that has lower chance and is generally more dangerous than autodoc one. Possibly some of the most advanced CBMs could be autodoc-only or require other kind fo assistance (an NPC skilled in medicine?) as well to have a chance to be installed succesfully.

**Comment by Blank99** [comment]
Currently:
It is my opinion that autodocs are currently too obscure for the removal of manual installation by default to be a good idea.
They are also in too few locations, it would be far too easy for the player to be locked out of them entirely for the duration of the early and mid game, and while this can also happen to other aspects of the game, this is a lot more unlikely.
This is not only a concern for simple/minor CBM, but also for cyborg professions if the change also removes CBM removal.

For the future:
For the current fluff of CBMs, what little there is, manual installation and installation with non-Doctor NPC aid should still be possible, even if under a different success rate.
Fluff changing and things like deepening the system to include slots for swappable bionics would, of course, alter this answer.


Regarding the general discussion:
I see a lot of talk of classifying CBMs as self-installable or not, but it concerns me that nobody is making allowances for non-regular-human characters, there's also flaws in some of the proposed rule of thumb classifications.
Self-trepanation sounds impossible, but what if you are a slime? A rule of thumb about not allowing self-installation on a CBM that occupies both hands sounds reasonable, but what if you have an NPC giving you a hand? What if you have more than two hands? What if the CBM is Integrated Toolset, and thus after installing half of it in your off-hand, it helps installing the other half on your main hand?
Replacing your eyes is one thing, but gluing retractable sunglasses to your face is quite another, despite both sharing the same slot. Similarly, it's one thing to stick an adrenal pump into your guts, it's another to replace your stomach.

And what difference would it make to have the sensory dulling CBM or mutating all the way into Prototype and being unable to feel pain?

**Comment by Night-Pryanik** [comment]
> They are also in too few locations

Do you think they should be in every building in the game? How many locations do you think it would be not "too few"?

> it would be far too easy for the player to be locked out of them entirely for the duration of the early and mid game

CBMs are not for early-game. Early-game player just don't have enough skills to reliably install bionics, most likely won't find anesthesia etc. Mid-game player would surely find one or more hospitals and is able to clean it. I'm not seeing any problem for mid-game player here.

> flaws in some of the proposed rule of thumb classifications.

This rule of thumb should be taken into account AFTER deciding how to deal with the need of some sort of anesthesia for medical self-operations (characters with no pain feeling mutations or sensory dulling CBM should be excluded, of course). Local anesthesia should be suitable for only the simplest of the CBMs, and general anesthesia is too seriously affecting your consciousness to make self-installation viable.

All of your examples (like slime, more than two hands etc) wasn't covered by the manual installation either. The problem of CBMs interacting with mutations is a huge one, and is out of scope of the question of this issue. If we decide to remove manual installation by default, we always could permit to self-install for these edge cases later.

**Comment by Blank99** [comment]
> Do you think they should be in every building in the game? How many locations do you think it would be not "too few"?

Once every two/three average towns, so even under bad luck the failed cyborg doesn't have to run too far from the initial revealed map to be able to save themselves. Alternatively, Labs spawn pretty reliably, so adding them there or the ability to turn dissectors into CBM installers would add a reliable but decently-guarded option. Labs are also chockful of failed cyborgs and you can start as one there, so it's a fluff-compliant solution.
There's also the concern of Hulks bringing down buildings by the time you have the CBMs you want to install. In general, even high level CBM installing shouldn't be harder than high end deathmobile-making or high end mutations.

> CBMs are not for early-game. Early-game player just don't have enough skills to reliably install bionics, most likely won't find anesthesia etc. Mid-game player would surely find one or more hospitals and is able to clean it. I'm not seeing any problem for mid-game player here.

Several CBM are pretty much low level crap that are barely useful in the early game at most, several starting professions require early access to CBM removal or are set up so you could make early installs easily.
Things introduced in these changes like the reliance on anesthesia are problematic with these, so I'd rather have alternatives. I have no problem with alternatives being higher risk or limited in some way.

> This rule of thumb should be taken into account AFTER deciding how to deal with the need of some sort of anesthesia for medical self-operations (characters with no pain feeling mutations or sensory dulling CBM should be excluded, of course).

Agreed, tho with the caveat that NPCs with decent ranks in the relevant skills should also count as an alternative.

> All of your examples (like slime, more than two hands etc) wasn't covered by the manual installation either. The problem of CBMs interacting with mutations is a huge one, and is out of scope of the question of this issue. If we decide to remove manual installation by default, we always could permit to self-install for these edge cases later.

This is more of a tangent to the manual installation discussion, but while what we have doesn't account for these modifications, it doesn't restrict things except for anesthesia either. Allowing too much is less frustration-inducing for the player than allowing too little.

All the exceptions don't need to be added from the start, but they should still be part of the design outline.
Designing new restrictions for the system without thought for existing body modifications and then adding later the exceptions would lend itself to them being added as patches and hacks, rather than arriving at a more elegant and extension-friendly solution from the get go.

**Comment by TechyBen** [comment]
The "broken cyborg" problem could be averted by a non-surgical option to "nullify" the leaky/sparky/faulty CMBs. It would only turn them into CMBs that do nothing... so improving survival rate, but still needing a auto-doc to fully remove.

**Comment by nexusmrsep** [comment]
A side idea that came to my mind - it's difficult (if not impossible) to do a self-instalation, but with anesthesia kit and good first-aid skill, you can perhaps manualy install bionics on willing NPC's?

**Comment by ZhilkinSerg** [comment]
> with anesthesia kit and good first-aid skill, you can perhaps manualy install bionics on willing NPC's?

With a long rope on unwilling too.

**Comment by kevingranade** [comment]
I feel like I need to reiterate this, because it seems to have been overlooked:
> in the main game your options are autodoc or a npc doctor (or another TBD third party).

The context of the discussion around when to allow self-installation is what a mod should allow, with default settings the answer is "never".

> it concerns me that nobody is making allowances for non-regular-human characters,

This is normal, it's up to people who are maintaining mutations to propose and make changes to integrate mutations with surgery.

> Once every two/three average towns, so even under bad luck the failed cyborg doesn't have to run too far from the initial revealed map to be able to save themselves.

We're not going to guarantee that, "if you search x towns you'll find the thing you're looking for",  just like we don't guarantee that really bad day is survivable or that it's possible to escape from a prison or lab.  If you don't like your odds, don't start as a failed cyborg.

> Agreed, tho with the caveat that NPCs with decent ranks in the relevant skills should also count as an alternative.

This is about self-installation.  Having a NPC do it is perfectly fine, though at the moment it is only supported for mission-related NPCs.

> Allowing too much is less frustration-inducing for the player than allowing too little.

The way we are doing it is very intentional, if we start out more restrictive, then adding features is acceptable.  If we make it permissive to start with we have to argue and justify every step of the way (see threads like this for example).

>  a non-surgical option to "nullify" the leaky/sparky/faulty CMBs. 

Good idea, I'm in favor.

> with anesthesia kit and good first-aid skill, you can perhaps manualy install bionics on willing NPC's?

Certainly.

**Comment by Blank99** [comment]
> We're not going to guarantee that, "if you search x towns you'll find the thing you're looking for", just like we don't guarantee that really bad day is survivable or that it's possible to escape from a prison or lab. If you don't like your odds, don't start as a failed cyborg.

It's not about guaranteeing it, that's plain unlikely with the way mapgen works now, it's about what is _likely_. There's no guarantee that the lab will not generate you in a room where the only exit is a room with a turret, for example, but it is, in general, unlikely to happen, you'll start a few dozen lab starts before ever seeing that or a case where the surface is not reachable.
Upping the number of potential autodoc places, toying with the spawning possibilities of these places, or adding one that appears with considerable regularity but in exchange is more protected would move the failed cyborg situation towards 'likely to be technically possible'.
Alternatively, letting the failed cyborg patch those failed bionics in some way so they don't kill them even if they can't remove them would achieve the same goal.

> This is about self-installation. Having a NPC do it is perfectly fine, though at the moment it is only supported for mission-related NPCs.

I'm not talking about perfectly or a dedicated NPC, I'm talking about toying with 'self installation' to check for regular NPC helpers on it's code that decides whether you can or cannot self-install something, and then using their skill level, either alone or together with the player's depending on the situation, to decide your chances of success.

Hell, combine it with the 'install on willing NPCs' idea and you could cultivate a mad doctor NPC by having them around when manually installing CBMs, thus making them practice their skills on 'volunteers', before being confident enough to let them put you under the knife.

**Comment by kevingranade** [comment]
> I'm not talking about perfectly or a dedicated NPC, I'm talking about toying with 'self installation' to check for regular NPC helpers on it's code that decides whether you can or cannot self-install something, and then using their skill level, either alone or together with the player's depending on the situation, to decide your chances of success.

> in the main game your options are autodoc or a npc doctor (or another TBD third party).

**Comment by TechyBen** [comment]
Yeah, for now the self instal is going to have to be demoted to mod status if ever.

I wonder about those options (CMB vending machines where you place arms/head in for a simple CMB install :dagger: ). But that is for another thread...

As Kevin mentioned. An option for "tinkering" (or whatever we label it as) to fix broken CMBs can be added to give options to the cyborg character starts. These quick fixes won't be as simple or complete as an autodoc, but would give you an option for survival until you find one (should city gen make it take 100 days etc to find).

Something like a makeshift anesthetic kit (+ possibly a dead bionic)+ skills + tools = turning "leaky/sparky/noisey" CMB into "broken". Turning it from a near death dealing CMB to a simple debuf/stat changing CMB. This could also offer mod routes for swappable CMBs in the future (swappable sockets/slots if already fitted, finger pick for finger lighter, night vision for heat vision swaps etc). While also *not* changing the limit of installs requiring autodocs or NPC docs.

(At least I hope my idea would work that way. As said, I'm just looking to tackle the change to game mechanics of cyborg starts, that the Autodoc causes. While not touching the balance of CMB installs suggested above.)

**Comment by Night-Pryanik** [comment]
I think I'll try to add either a new private bionic clinic or another variant of doctor's office with an autodoc. I hope this (alongside with three other already implemented autodoc spawn locations) could satisfy those who thinks that autodoc is too hard to find.
I also planned to add another location for autodoc to spawn in labs, but I prefer to wait until #24114 is finished and merged.

**Comment by CoroNaut** [comment]
Is there any thought given to making the autodoc a vehicle mountable item? I would like to be able to have easy access in the late game. If it's a multi-tile structure then it will reinforce the idea that it would be very difficult to rebuild it.

**Comment by TechyBen** [comment]
> Is there any thought given to making the autodoc a vehicle mountable item? I would like to be able to have easy access in the late game. If it's a multi-tile structure then it will reinforce the idea that it would be very difficult to rebuild it.

Personally, if I'm gonna get that far late game I'm going to just RP it and delete it from the hospital, and cheat it into the base/area I'm in to say the char "dragged" it across. ;)

**Comment by FulcrumA** [comment]
I don't think deconstruction/construction of autodoc should require high mechanics skill. It seems a rather straightforward thing compared to more advanced parts we already have if it's not about building it frm scratch. I'd rather imagine certain computers/first aid skill to operate it but if someone found one, but scavenging and hooking it up elsewhere should be doable with basic mechanics/electronics skill. If we'd consider it, building one from nearly scratch should be based on relatively high electronics and considerafle first aid too (at this point though, I have to nitpick that "first aid" seems to be a rather unfitting name for what seems to be including quite specialistic medicine knowledge).

**Comment by Firestorm01X2** [comment]
Anesthesia kit should be craftable with enough skills. And it should not be very hard. Main problem should come from necesserity of finding ceratin drugs for it.

**Comment by papersplease** [comment]
@Firestorm01X2 That's a matter of what one wants from the CBM progression. Uncraftable and rare anesthetics = every implant is valuable, installation rate is fairly limited; but CBM installation is not easily available at the start. Anesthetics craftable from common components = installation rate is essentially unlimited as one can find any existing drugs in arbitrary quantities already; but CBMs become available at the start. The second variant isn't very good I guess, because it incentivizes players to hoard CBMs and install them in packs.

IMO the anesthetic or the components should be approximately as rare as a stethoscope, regardless of being craftable, and installation should also require some skills. If craftable, making one shot should take a long time, probably several hours. That way the player will be able to smoothly progress through different tier CBMs from the start.

**Comment by Firestorm01X2** [comment]
@papersplease 
>That's a matter of what one wants from the CBM progression. Uncraftable and rare anesthetics = every implant is valuable, installation rate is fairly limited; 

At this rate we can easily remove some bionics relatively useless bionics.
Also uncraftable and rare anesthetics is not particulary realistic especially if we took into accunt what player able to craft in other areas.

Also we already have CBM slots for limitation as far as I know. Let's use that.

**Comment by Night-Pryanik** [comment]
> if we took into account what player able to craft in other areas.

We shouldn't take that into account.

> Also we already have CBM slots for limitation as far as I know. Let's use that.

CBM slots system is an unfinished feature, so before using it for any means it should be finished first.

**Comment by Firestorm01X2** [comment]
>We shouldn't take that into account.

Then we should so sanity check. I bet that it is possible to create anastetics IRL using proper meds.

**Comment by nexusmrsep** [comment]
By proffesional pharmaceutical companies, not by anesthesiologists in their basement chem labs Breaking Bad style.

**Comment by Night-Pryanik** [comment]
Using autodoc require way less skills than making anesthetics. High levels in first aid (as an substitute for knowledge of human anatomy and physiology) and cooking (as chemistry), perhaps 7.

**Comment by nexusmrsep** [comment]
What I want to say is there are chems that you would not get at your garage even if you tried, or made unclear version after which you just might not wake up ever. "Magic" here and there is not a reason for magic in other places, aspecialy if balancing reasons come to effect.

**Comment by Firestorm01X2** [comment]
>What I want to say is there are chems that you would not get at your garage even if you tried, or made unclear version after which you just might not wake up ever.

I can say same things about craftale mutagens. But somehow we can live with that.

**Comment by nexusmrsep** [comment]
Of course we can live with that. Its an scifi thing, with no reference to RL. You wanna talk about comparisons about laser rifles and rifles? Possible only by lore, not by IRL referencing.

**Comment by Night-Pryanik** [comment]
8 cooking is ok for mutagens, and probably ok for anesthesia crafting. But I made so many places when one can find anesthesia. Is it really needed to be craftable?

**Comment by sfsworms** [comment]
I think it should. With a high skill requirement it won't be easy in the mid-game anyway. Being frustrated by having big batches of bionics you can't install sounds bad.

**Comment by Firestorm01X2** [comment]
@nexusmrsep 
>Of course we can live with that. Its an scifi thing, with no reference to RL. 

And what about car construction then?

**Comment by nexusmrsep** [comment]
> And what about car construction then?

Good example: you can build a car from parts, as some mechanics do, but you won't be able to make an engine block by yourself.

Let's take that principle to anesthesia kit:
I believe, like @Night-Pryanik said, that it might be craftable from existing drugs with high cooking, but not from scratch. I simply ask myself if there are in-game drugs that with careful mixing would result with something like anesthesia. It's not as simple as adding meth to aspirin and mutated poppy juice. At least I can't exactly convince myself. Notice that at the begining of the discussion on autodoc I was for making it craftable myself, but I'm not that sure after all that had been said afterwards.

**Comment by Firestorm01X2** [comment]
>8+ cooking is ok for mutagens, and probably ok for anesthesia crafting.

I think it should slightly lower.  It is genetics vs anesthesia. 5, 6 or 7 probably.

>Is it really needed to be craftable?

Why not? It is pretty logical that player with highest medical and cooking skill could learn ways to craft anesthetics since some hardcore genetics is available.

Also if there is really need to restrict CBM installation then it is easier to make CBM hilariously harder to install or find. This way dancing around autodoc is just harder way to do it.

**Comment by Night-Pryanik** [comment]
I'm not against making anesthesia kit craftable, I'm against the conception that everything must be craftable because "why not?" Either way, I'm not a chemist or anesthesiologist, so I don't know what components are needed to craft surgery-fitting general anesthesia and in what quantities. Anyone more competent than me on this matter are welcome.

> zombies drop only rags with some rare exceptions

Why should they? People can die without their clothes getting tattered to rags, and something tells me this should be usual behavior, not the rare case. _/closing offtopic_

**Comment by nexusmrsep** [comment]
> because "why not?"

Actually a complicated or rare component recipe gives a lot of incentive for the players to explore and to act. They can either find anesthetic kit or the components themselves, either way it pushes them to act, adding momentum to their gameplay.

**Comment by Firestorm01X2** [comment]
>Actually a complicated or rare component recipe gives a lot of incentive for the players to explore and to act. 

And why it is not CBMs itself in the first place?

**Comment by Night-Pryanik** [comment]
> Hell, you even can craft CBM itself. But not atastesia kit to install it.
Consistency? No?

Have you considered the fact that players were given too much possibilities (like crafting CBMs) in the past could be a bad thing? That decision to make CBMs craftable were probably given the same "why not?" rationale and was not based on the well-thought design convention? That maybe we should remove the craftability of CBMs for the sake of consistency and not make anesthesia craftable?

**Comment by Firestorm01X2** [comment]
>Have you considered the fact that players were given too much possibilities (like crafting CBMs) in the past could be a bad thing?

Maybe. But moving this way lead us to remove features from the game.

Anyway now we have CBMs lying around with rare kit to install it. Solve it somehow.

**Comment by Night-Pryanik** [comment]
> rare kit

The kit has guaranteed spawns at some places, in other it has rather high chance to spawn. The places where it spawns could probably be found in most medium to large cities. Are you sure it isn't "lying around" like CBMs do, and that it is indeed rare?

**Comment by Night-Pryanik** [comment]
> How much per city in average?

I started several new fixed scenario play nows with default city sizes and distances. I used `Reveal map` debug option to see the neighborhood. I got 1-2 hospitals, 3-11 veterinarian offices, 4-9 doctor's offices, 2-3 military bunkers (with potential of spawning a bunker with an infirmary), and 50-80 house basements (with a potential of several of them could be the bionic basements). And that's just per one overmap. Almost every new average size city means doctor's or veterinarian's office, and that means another anesthesia kit.

> How is it in comparsion of welding rig?

It's a bad object for comparison, so I won't compare them.

> Also what about failing installation. Player may waste kits.

So what? Who tells that installing bionics is an easy job?

**Comment by Firestorm01X2** [comment]
>I started several new fixed scenario play nows with default city sizes and distances. I used Reveal map debug option to see the neighborhood. I got 1-2 hospitals, 3-11 veterinarian offices, 4-9 doctor's offices, 2-3 military bunkers (with potential of spawning a bunker with an infirmary), and 50-80 house basements (with a potential of several of them could be the bionic basements).

If it gives around 10 kits then it not that bad.

How about hand made kit made of normal kit. Lets called it: 
Deluted anesthesia kit. It may come as 2 or 3 items per normal kit.

**Comment by sfsworms** [comment]
I like the idea of the diluted kits.

I also think that the restriction step should be at the CBM availability rather than at the kit level, in part because it then let people had more use for anasthesia kits (fixing broken limbs with autodoc? Surgically removing some mutations? Operating on NPC to install remotely detonated bombs in their thorax?) without shifting the game balance to much. Surgery can be a neat mechanic in its own right apart from CBM. 

I'm also not a fan of nerfing CBM generally, but then that is more of a personnal opinion, I've always liked the more cyberpunky feel of CDDA, but I can understand that it's going in another direction if that's what the devteam wants.

**Comment by kevingranade** [comment]
If you want craftable anesthetics, find some sources outlining how to make them.  
If you want these kids to be usable by autodocs, the recipe needs to be for a modern anesthesia.  
If you want to support diluting anesthetics, find a source outlining how this works.  
Until you've done that I don't care what other arguments you've made, game balance arguments do not trump needing an IRL rationale for craftability of IRL items.

**Comment by BrianLefler** [comment]
https://www.quora.com/How-do-you-make-sodium-pentothal

- sulfur +hydrogen => hydrogen sulfide (sewer gas) [at 450 °C]
- hydrogen sulfide (sewer gas) + calcium cyanamide (fertilizer) => thiourea
- grapes or strawberries => diethyl malonate [via purification]
- thiourea + diethyl malonate => sodium pentothal

Sodium Pentothal is less preferable than Propofol for anesthetic (dunno way), so it'd be fair to call the item "homemade anesthetic" and give the formal chemical name in the description.

So a recipe could be one step:
- sulfur + hydrogen + fertilizer + grapes/strawberries = homemade anesthetic

Or two:
- sulfur + hydrogen = swamp gas
- swamp gas +  fertilizer + grapes/strawberries = homemade anesthetic

I don't know what quantities you'd want to use for the reagents.  All the recipes would require a chemistry lab.

**Comment by nexusmrsep** [comment]
There is another thing that you might need to address regarding installation of bionics. I don't remember it it was already mentioned, so excuse me if I repeat this topic.

Lore says CBM's come in a ready-to-install canisters, but its also fairly possible to rip out some bionics form bodies. It's quite unlikely that these are as easy to install as a factory prepared kit (that might contain some helping tools for the installation).
One way of implementing this feature would be to raise used-CBM's requirements or lower new-CBM's requirements for installation (or both - slightly).
Another approach would be locking used-CBM for autodoc instalation only, while new-CBM's would be viable for self installation too.
As for code-wise method I'd think of a "USED" flag that can be added to all CBM's butchered from bodies, that would be afterwards be recognized by all available methods od installation to vary the outcome.

**Comment by kevingranade** [comment]
We've certainly floated the idea of requiring cleaning, sterilization, and possibly repair of extracted CBMs.
I think the best way to handle this is to make it un-installable until you do the proper prep.

**Comment by FulcrumA** [comment]
Maybe some sort of counterpart to clothes' "filthy" tag could be used. Cleaning and sterilization could be rolled up together and be done with throug the use of disinfectant-soaked cotton balls/rags. 

Bonus balance points in the fact that while in the beginning of the game disinfectant is prized and vital resource, from mid-game on (around the time new CBMS are being acquired and installed) average player is more resistant to infection and/or strong and equipped enough that they give opponents less chances to infect them - so this would even out usefulness of disinfectant items thorough the game.

**Comment by Night-Pryanik** [comment]
I fear that players will perceive the requirement of using disinfectant to disinfect salvaged CBMs just as another boring chore.

And yet I still think CBMs (most if not all) couldn't be butchered from corpses undamaged and in ready-to-install state. I'd remove the ability to salvage CBMs from corpses, leaving only the simplest and/or plausible. But that's just my opinion.

**Comment by FulcrumA** [comment]
I don't really think it'd be that much of a chore. Unless someone butchers and succesfully extracts CBMs en masse (a very unlikely thing) and somehow feels the need to install every one of them (with mod that'd allow them to install every one of them after extraction I guess), they will only choose and disinfect something from time to time. Activating disinfectant and choosing an item rather than a bodypart should be merely a few keypresses all the same, sure.

Special creatures corpses being a good source of CBMs is a decent feature in general. If it really needs balancing (and I don't think it does as it's not merely the matter of getting a CBM but also succesfully installing it) perhaps a chance of them being some burn-out ones should be higher, with some more books offering some more recipes (manuals in hospitals regarding theory of creation of some of the most common models, FEMA/bunker/outpost/other-soldier-location documentations for military-grade augmentations and so on). 

Alternatively, making CBMs depend on some biological features (can't install hydraulic arms without bone and muscle structure developed enough to support said arms when they're inactive, for example). Would require a few more intermediate/weaker CBMs to allow certain improvements in some areas for characters not fitting the profile.

**Comment by nexusmrsep** [comment]
> I fear that players will perceive the requirement of using disinfectant to disinfect salvaged CBMs just as another boring chore. And yet I still think CBMs (most if not all) couldn't be butchered from corpses undamaged and in ready-to-install state. I'd remove the ability to salvage CBMs from corpses, leaving only the simplest and/or plausible. But that's just my opinion.

I find the possibility of getting CBM's from corpses a fun and rewarding feature, and it's removal im my view would have much more negative impact, than for example implementing a simple rule that you can't self-install a second-hand CBM, and always have to find an autodoc for it. 

> We've certainly floated the idea of requiring cleaning, sterilization, and possibly repair of extracted CBMs. I think the best way to handle this is to make it un-installable until you do the proper prep.

The possibility of getting burnt-out bionics is in my opinion enough, but i agree that there is no middle state like a CBM with a CBM_DAMAGED flag, that could be repaired using similar (but less) requirements as for building one. We only hae a 0/1 outcome. You can also manipulated the statistics here - for ex. higher butchering skill = better chances of getting intact CBM (but still with CBM_USED flag). Also you can lower the chance of burnt out bionic in favor of CBM_DAMAGED. In that way unskilled players will either hoard those for late-game or leave behind like they would with burnt-out bionics, so the impact would be negligible, if not slightly beneficial all in all.

Even if both flags (or similar application) - CBM_DAMAGED & CBM_USED would be applied to CBM's salvaged from bodies, and second-hand CBM's would need preparations (disinfecting) I don't think it would be an extreme chore, and I agree with @FulcrumA on that matter. Using disinfectant in late game is a nice idea.

**Comment by Night-Pryanik** [comment]
> implementing a simple rule that you can't self-install a second-hand CBM

I thought it was clear that you can't self-install any CBMs, except for the (not yet existing) mods.

**Comment by nexusmrsep** [comment]
> I thought it was clear that you can't self-install any CBMs, except for the (not yet existing) mods.

My misunderstanding, sorry. Was having in mind that in the opening issue you said...
> Bionics (for now) still can be installed old-style manual way.

... but I forgot that @kevingranade said along the way that this option was wasn't going to be left open,  so I thought that we still discuss potential options of viability of self-installation (and possible conditions on that). So I guess disregard everything I said there that relates to self-installation. Perhaps a mod-designer might consider those options though.

**Comment by Night-Pryanik** [comment]
> I don't really think it'd be that much of a chore.

Before: butcher -> get CBM -> install. After: butcher -> get CBM -> disinfect -> install. This extra step will be considered as tedious chore (based on what players said about extra step of washing filthy clothes when it was implemented).

And I just want to remind that this issue is about manual installation of bionics. For discussion on the future of salvaged bionics one can look at #20383.

**Comment by mlangsdorf** [comment]
While all these issues are being sorted out, can we get an incentive to use the auto-doc? It's disappointing to go around finding auto-docs and anesthesia kits, clearing the area around the auto-doc and distracting hordes so they'll stay away, making you way to the auto-doc, and then realizing you could have just injected yourself with a CBM from your home base with 97% less hassle and no worse chance of successful installation.

**Comment by FulcrumA** [comment]
@Night-Pryanik 
>This extra step will be considered as tedious chore (based on what players said about extra step of washing filthy clothes when it was implemented).

Compared to what? Actually having to repair and clean CBM through several different actions (given that's what is being considered to differentiate between salvaged and ready-to-use ones)? Keeping that in mind I'd consider merely having to use disinfectant a pretty simple and elegant solution, hardly a chore, a thing that wouldn't require too much of additional code and with aforementioned bonus of balancing disinfectant in a way that it stays roughly as useful thorough the game, even if not always in the same role.

On top of that, I again have to underline that compared to clothes to be cleaned in the early to mid game, CBMs are much more rare and so any cleaning will be also a much more rare occurence, compared to chores on the level of refilling gas tank of a car and possibly, given how mobile huge part of the playerbase is, even more rare than that.

**Comment by Night-Pryanik** [comment]
@mlangsdorf I think the required minimum for leaving only the autodoc and NPC surgeon is accumulated. So I'm waiting for lab jsonification project by @BrianLefler to be merged (adding another spawn of autodoc and anesthesia kit), and then the plan is to disable the manual installation by removing `"use_action": "install_bionic"` field from CBMs items. The code for manual installation will still be here for some time for any modder who wants to return it in the game in mod form.

**Comment by Night-Pryanik** [comment]
If I understand you correctly, this installer would still allow self-installation, with some new added functional? In that case I say "No" because if there are some highly advanced nano-robots or some other magic exist in this installer, this is NOT a self-install, and if I remember correctly, we don't have such an advanced technology in our lore.
I could give self-installation a slightest chance to exist only if it would be majorly reworked first in terms of time consuming and the need of anesthesia.

**Comment by BrianLefler** [comment]
Yes, that's correct.  I can try to make the action consume time and cause a some amount of damage to use; whatever painkillers/anesthesia is required would be part of the kit.

The lore I can see, at least in the wiki, says: "CBMs are surgery kits that contain all the tools required to install a cybernetic implant, also called bionics, in a human. They are automated in some ways, but manufacturers' warranties insist on having a fully qualified technician and medical doctor to assist in the installation--in a proper medical facility." This would preserve that lore, but separate the surgery kit from the bionic implant, and make the kits more rare since most installs would be happening in a medical facility via docs/autodocs.

I have always sort of imagined the installer as effectively a one-shot specialized surgical robot with a lot of manual steps to position & program it (thus requiring the first aid skill).  There's no nano-robots in lore to my knowledge; but there are definitely robots.  But I guess the nature of a CBM installer does not have to be really decided until someone wanted to create a CBM recipe, and I know @kevingranade had ideas on that.

Thanks for the feedback.  Other opinions?

**Comment by ZhilkinSerg** [comment]
We can always put new kit into BrightNights mod (which is more sci-fi than vanilla).

**Comment by Night-Pryanik** [comment]
CBMs canisters do not contain something beside the CBM itself. No tools, no anesthesia, painkillers, bandages or any other medical stuff, and certainly no robots of any sort. It's just bionic implant in there, that's all.

**Comment by sfsworms** [comment]
Where is the "official" lore on CBMs then?

**Comment by kevingranade** [comment]
I'm not interested in having any kind of standalone installer in the main
game, if you want to add that as a mod, go crazy.
I don't anticipate anything other than a full surgical suite allowing
bionic installation, with the role of the surgeon being filled by either a
robot or a NPC.

I did think of one strange exception that doesnt really act like a bionic.
Some devices might just be implanted, like a tracker chip.  That doesnt
require surgery per se.

**Comment by Night-Pryanik** [comment]
> Where is the "official" lore on CBMs then?

AFAIK we don't have any "official" lore, and this is the reason for many inconsistencies and incorrect conclusions.
